upgrade_logic/base_logic.py

```diff
@@ -55,13 +55,20 @@ def __function_is_relevant(
 
 # v1 newer than v2 (e.g. 1.1.2 > 1.1.1)
 def is_newer(v1: str, v2: str) -> bool:
-    a = [int(x) for x in v1.split(".")]
-    b = [int(x) for x in v2.split(".")]
+    a = [__get_int_from_string(x) for x in v1.split(".")]
+    b = [__get_int_from_string(x) for x in v2.split(".")]
     if len(a) != len(b) and len(a) != 3:
         raise Exception("invalid version format")
     return __is_newer_int(a, b)
 
 
+def __get_int_from_string(value: str) -> int:
```

Comment:
still getting some error in the console when i first start it (dev setup with the exclusion of the services)

<img width="974" alt="Screenshot 2024-04-10 at 17 59 09" src="https://github.com/code-kern-ai/refinery-updater/assets/104008550/01bc0336-cd98-4f77-9583-768872e61eeb">